### PR TITLE
HTML-CSS-PopUp

### PR DESCRIPTION
**Pull request by NastjonkaK**
### HTML-CSS-Popup

[Demo](https://nastjonkak.github.io/PopUp/) |
[Code base](https://github.com/NastjonkaK/PopUp)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by Temu4** [comment]
@NastjonkaK , nice work!
But, there are a few moments (not a code rewiev, just visual rewiev):
- make all interactive elements available from the keybord

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            		</li>
+            		<li class="popupIconsItem">
+	            		<a href="#">
+	            			<img src="Icons/alarm.png">
+	            			<div class="captions"> Alarm </div>
+	            		</a>
+            		</li>
+            		<li class="popupIconsItem space"></li>
+            	</ul>
+            		<input type="checkbox" id="button_more" class="MoreApps">
```
Comment (Temu4) [inline submissions/NastjonkaK/HTML-CSS-Popup/index.html]: Also you can avoid useing of `id` and make your cade more compact as that:
```html
<label class="MorePopUp">More 
   <input type="checkbox" class="MoreApps">
</label>
```

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            </div>
+			<li class="menu-item"> <img src="Icons/bell.png"></li>
+			<li class="menu-item"> <img id="ava" src="Icons/avatar.png"></li>
+			<div id="triangle">
+				<div id="empty"></div>
+			</div>
+		</ul>
+	</nav>
+</body>
+</html>
```
Comment (Temu4) [inline submissions/NastjonkaK/HTML-CSS-Popup/index.html]: Please, add an empty line to the code.

File: submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+	margin-left: 20px;
+}
+
+.menu img,
+.button {
+	width: 24px;
+	height: 24px;
+}
+
+#ava {
```
Comment (Shramkoweb) [inline submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css]: Лучше не стилизовать по тегу, так как потом будет тяжело перебить стили.

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	            		<a href="#">
+	            			<img src="Icons/chat.png">
+	            			<div class="captions"> Chat </div>
+	            		</a>
+            		</li>
+            	</ul>
+            </div>
+			<li class="menu-item"> <img src="Icons/bell.png"></li>
+			<li class="menu-item"> <img id="ava" src="Icons/avatar.png"></li>
+			<div id="triangle">
```
Comment (Temu4) [inline submissions/NastjonkaK/HTML-CSS-Popup/index.html]: Also, more "canonical" is useing `::after` or `::before` css pseudo-classes for such porposes.

File: submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+.button {
+	margin-left: 20px;
+ 	background-image: url(../Icons/dots.png);
+ 	background-size: cover;
+ 	cursor: pointer;
+}
+
+.menu li:hover {
```
Comment (Shramkoweb) [inline submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css]: Вложенность, не лучшее решение.
Лучше добавить класс на li и стилизовать по нему.
Почитай про BEM

File: submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+.popupIconsItem:hover {
+	outline: 1px solid #a1a1a1;
+}
+
+.space:hover {
+	outline: none;
+}
+
+.popupIcons a,
```
Comment (Shramkoweb) [inline submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css]: У тебя в некоторый местах классы именуются в camelCase, а потом в snake-case.
Опять же почитай про BEM =)

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html (hunk 4 above)
Comment (Shramkoweb) [inline submissions/NastjonkaK/HTML-CSS-Popup/index.html]: У списка ul дети могут быть только li.
Проверь верстку на валидатор: https://validator.w3.org/nu

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+            		<li class="popupIconsItem hidden">
+	            		<a href="#">
+	            			<img src="Icons/chat.png">
+	            			<div class="captions"> Chat </div>
+	            		</a>
+            		</li>
+            	</ul>
+            </div>
+			<li class="menu-item"> <img src="Icons/bell.png"></li>
+			<li class="menu-item"> <img id="ava" src="Icons/avatar.png"></li>
```
Comment (Shramkoweb) [inline submissions/NastjonkaK/HTML-CSS-Popup/index.html]: У картинок лучше не пропускать атрибут alt

**Comment by Temu4** [comment]
If you have some questions or issues, please ask here 🐱

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html (hunk 1 above)
Comment (YanaGushlevskaya) [reply submissions/NastjonkaK/HTML-CSS-Popup/index.html]: @NastjonkaK Сontinuing to talk about label/input -  I was recommended to watch [this](https://www.youtube.com/watch?v=E6kLaaQFctU&feature=youtu.be) video, when doing this task. Now I recommend you, it is really useful.

**Comment by YanaGushlevskaya** [comment]
@NastjonkaK Hi, great job! I'll add a few comments)

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+	<title>PopUp menu</title>
+	<link rel="stylesheet" href="Style/style.css"/>
+</head>
+<body>
+	<nav>
+		<ul class="menu">
+			<li class="menu-item"> Mail </li>
+			<li class="menu-item"> Photos </li>
+			<input type="checkbox" id="ButtonPopUp" class="popUp">
+            <label for="ButtonPopUp" class="button"></label>
```
[the remark this answers is not in the document]
Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/index.html]: Thanks for the tip!

File: submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css (hunk 3 above)
Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css]: Дякую! 
Чому буде важко перебити стилі, використовуючи id?

**Comment by NastjonkaK** [comment]
> Also as I understood, when your element is focused, the gray border (hover effect) shoudn't be dispalyed, only blue border/outline.

@YanaGushlevskaya You're right. I didn't notice it. Thank you! :)

**Comment by NastjonkaK** [comment]
> * make all interactive elements available from the keybord;

@Temu4 Do you mean using "tabindex" attribute?

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html (hunk 1 above)
Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/index.html]: > Also you can avoid useing of `id` and make your cade more compact as that:
> 
> ```
> <label class="MorePopUp">More 
>    <input type="checkbox" class="MoreApps">
> </label>
> ```

Then the "More" button stops working. Why?

Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/index.html]: > Also you can avoid useing of `id` and make your cade more compact as that:
> 
> ```
> <label class="MorePopUp">More 
>    <input type="checkbox" class="MoreApps">
> </label>
> ```

Then the "More" button stops working.

File: submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css (hunk 3 above)
Comment (Shramkoweb) [reply submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css]: https://zen.yandex.ru/media/id/5a8ed6eddcaf8e23b97cf564/osnovy-css-vesa-selektorov-5ad4e488830905f568348f74
You can read this

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html (hunk 2 above)
Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/index.html]: Done.

File: submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css (hunk 3 above)
Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css]: Thanks!

File: submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css (hunk 6 above)
Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css]: I changed all classes names in camelCase.

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html (hunk 4 above)
Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/index.html]: Thanks for a tip! I checked the code by validator and fixed the bugs.

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+            <div class="iconsContainer">
+            	<ul class="popupIcons">
+            		<li class="popupIconsItem">
+	            		<a href="#">
+	            			<img src="Icons/docs.png">
```
[the remark this answers is not in the document]
Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/index.html]: I've already fixed this bug.

File: submissions/NastjonkaK/HTML-CSS-Popup/index.html (hunk 4 above)
Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/index.html]: I don't think it's appropriate in this case.

File: submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css [hunk 10]
```diff
@@ -0,0 +1,192 @@
+body {
+	font-family: "Montserrat";
```
Comment (Shramkoweb) [inline submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css]: Will be good practice to add few web safe fonts
U can read this article http://codeguide.academy/html-css.html

Comment (NastjonkaK) [reply submissions/NastjonkaK/HTML-CSS-Popup/Style/style.css]: Thanks for additional information.

**Comment by dashakim** [comment]
Hello, good work! 
I'll give you advise as the same as you had given to me)
To check on validator page and fix errors.

**Pull request merged**